client/ayon_core/plugins/load/delete_old_versions.py

```diff
@@ -365,19 +394,40 @@ def main(self, project_name, data, remove_publish_folder):
         return size
 
     def load(self, contexts, name=None, namespace=None, options=None):
+
+        # Get user options
+        versions_to_keep = 2
+        remove_publish_folder = False
+        if options:
+            versions_to_keep = options.get(
+                "versions_to_keep", versions_to_keep
+            )
+            remove_publish_folder = options.get(
+                "remove_publish_folder", remove_publish_folder
+            )
+
+        # Because we do not want this run by accident we will add an extra
+        # user confirmation
+        if self.requires_confirmation:
+            contexts_list = "\n".join(sorted(
+                "- {folder[name]} > {product[name]}".format_map(context)
+                for context in contexts
+            ))
+            num_contexts = len(contexts)
+            s = "s" if num_contexts > 1 else ""
+            if not self.confirm(
```

Comment:
I would change `confirm` to private, and the method can expect only `contexts` and the rest of text would be filled there?

Reply:
I could do that - I just made it this way so that the `confirm` is generic and could potentially be reused down the line for other confirmations. Just felt the most sensible to have the confirm box be a generic function.

Happy to refactor it though if you still want me to. Let me know if you want that.

Reply:
Still `confirm` should be private method. I don't think there will be re-use of the method, but that still can be achieved by adding one other method `_confirm_delete` which would expect only contexts and prepare texts for `_confirm`.

Reply:
Done with https://github.com/ynput/ayon-core/pull/773/commits/1ed42d601a27b349e155ac541b84761b8e72799d